pkg/serviceprovider/access_token_mapper.go

```diff
@@ -54,23 +58,27 @@ func (at AccessTokenMapper) ToSecretType(secretType corev1.SecretType, mapping *
 	case corev1.SecretTypeDockercfg:
 		ret[corev1.DockerConfigKey] = at.Token
 	case corev1.SecretTypeDockerConfigJson:
-		// parse host from ServiceProviderUrl if that is not possible use the whole ServiceProviderUrl as host
-		host := at.ServiceProviderUrl
 		parsed, err := url.Parse(at.ServiceProviderUrl)
-		if err == nil && parsed.Host != "" {
-			host = parsed.Host
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse service provider url: %w", err)
+		}
+		host := parsed.Host
+		if host == "" {
+			host = at.ServiceProviderUrl
+		}
+		buff := new(bytes.Buffer)
+		conf := configfile.ConfigFile{AuthConfigs: map[string]types.AuthConfig{host: {Username: at.ServiceProviderUserName, Password: at.Token}}}
```

Comment:
I don't like `docker/cli` dependency. I would be ok-ish with `moby/api/types/registry/AuthConfig` https://github.com/moby/moby/blob/master/api/types/registry/authconfig.go#L16 but it has no `auths` map.
I would propose this-like function:
```
func (at AccessTokenMapper) encodeDockerconfig() (string, error) {
	encoded := base64.StdEncoding.EncodeToString([]byte(at.ServiceProviderUserName + ":" + at.Token))
	host := ...

	type Auths map[string]struct {
		Auth string `json:"auth"`
	}
	eh := struct {
		Auths Auths `json:"auths"`
	}{
		Auths: Auths{
			host: {
				Auth: string(encoded),
			},
		},
	}

	jsonBytes, err := json.MarshalIndent(eh, "", "\t")
	if err != nil {
		return "", err
	}
	return string(jsonBytes), nil
}
```

Reply:
+1, even if it is puts kind of support on us :)

Reply:
+1 for not using a dependency just because of a single struct that describes the format of a config file that is very unlikely to change.